gitdiff/patch_header.go

```diff
@@ -457,5 +458,30 @@ func parseSubject(s string) (string, string) {
 		break
 	}
 
-	return s[:at], s[at:]
+	return s[:at], decodeUTF8Subject(s[at:])
+}
+
+// Decodes a subject line if encoded using quoted-printable UTF-8 Q encoding. This format is the
+// result of a `git format-patch` when the commit title has an emoji (or other non-ASCII character).
+// See for reference: https://stackoverflow.com/questions/27695749/gmail-api-not-respecting-utf-encoding-in-subject
+func decodeUTF8Subject(encoded string) string {
```

Comment:
Let's call this `decodeSubject` so it could be expanded for more encodings, either now or in the future.